warehouse/accounts/services.py

```diff
@@ -363,9 +390,16 @@ def check_password(self, password, *, tags=None):
         return False
 
 
-def hibp_password_breach_factory(context, request):
-    return HaveIBeenPwnedPasswordBreachedService(
-        session=request.http,
-        metrics=request.find_service(IMetricsService, context=None),
-        help_url=request.help_url(_anchor="compromised-password"),
-    )
+@implementer(IPasswordBreachedService)
+class NullPasswordBreachedService:
+    failure_message = "This password appears in a breach."
+    failure_message_plain = "This password appears in a breach."
+
+    @classmethod
+    def create_service(cls, context, request):
+        return cls()
+
+    def check_password(self, password, *, tags=None):
```

Comment:
I think I don't quite understand under what circumstances we want to use this?

Reply:
We use it in local development, because we set all users to have the password "password", which appears in breaches. Thus without the ``NullPasswordBreachedService`` in local development, we would never be able to log into those users, and would have to generate secure password for them.